packages/react/src/useTable.tsx

```diff
@@ -205,8 +228,43 @@ export const useTable = <
         },
         sortable: "sortable" in field && field.sortable,
         relatedField,
-      } as TableColumn;
+        isCustomCell: false,
+      } satisfies RecordTableColumnValue;
     });
+
+    if (options?.columns) {
```

Comment:
The types in here are pretty wild, but do you think it would be feasilble to move this section to a helper function? This hook is starting to get quite big

Reply:
Moved the types into a separate file.